unity/Assets/Scripts/AgentManager.cs

```diff
@@ -905,12 +918,16 @@ private void ProcessControlCommand(string msg)
         this.activeAgentId = controlCommand.agentId == null ? 0 : controlCommand.agentId;
 
 		if (controlCommand.action == "Reset") {
+            this.agentState = AgentState.Processing;
```

Comment:
To not have to set the state like this for every action, maybe we should just set it outside of the `if` for the `string msg` version of `ProcessControlCommand`, and remove it from the others, minus the WebGL one which calls `ProcessControlCommand` not with a string due to the dynamic incompatibility.

Reply:
I just switched the AgentManager to use the ActionDispatcher and created a set of predefined agentManager action to clean that code up.